File: lib/src/main/java/com/sd/lib/cache/handler/impl/BytesHandler.java
package com.sd.lib.cache.handler.impl;

import com.sd.lib.cache.CacheInfo;
import com.sd.lib.cache.handler.BaseCacheHandler;

/**
 * byte数组处理类
 */
public class BytesHandler extends BaseCacheHandler<byte[]> {
    public BytesHandler(CacheInfo info) {
        super(info);
    }

    @Override
    protected byte[] valueToByte(byte[] value) {
        return value;
    }

    @Override
    protected byte[] byteToValue(byte[] bytes, Class<?> clazz) {
        return bytes;
    }

    @Override
    protected String getKeyPrefix() {
        return "bytes_";
    }
}

File: lib/src/main/java/com/sd/lib/cache/handler/impl/BytesHandler.kt
package com.sd.lib.cache.handler.impl

import com.sd.lib.cache.CacheInfo
import com.sd.lib.cache.handler.BaseCacheHandler

/**
 * 字节数组处理类
 */
internal class BytesHandler(info: CacheInfo) : BaseCacheHandler<ByteArray>(info, "bytes") {
    override fun encodeToByteImpl(value: ByteArray, clazz: Class<*>?): ByteArray {
        return value
    }

    override fun decodeFromByteImpl(bytes: ByteArray, clazz: Class<*>?): ByteArray {
        return bytes
    }
}